builtin/mv.c

```diff
@@ -292,8 +292,10 @@ int cmd_mv(int argc, const char **argv, const char *prefix)
 			continue;
 
 		pos = cache_name_pos(src, strlen(src));
-		assert(pos >= 0);
-		rename_cache_entry_at(pos, dst);
+		if (pos >= 0)
+			rename_cache_entry_at(pos, dst);
+		else if (!ignore_errors)
+			die(_("bad source: source=%s, destination=%s"), src, dst);
```

Comment:
Maybe we should say something along the lines that an entry already exists?

Reply:
`die(_("source already exists in the index under a different casing: source=%s"), src);` maybe? I could explain what to do, and that caseInsensitive option is true, but that seems too verbose for a relatively unlikely problem, especially as we aren't dealing with it optimally (we're still renaming on disk)

Reply:
I think a more correct patch would look like this:
  diff --git a/builtin/mv.c b/builtin/mv.c
  index 7dac714af9..8572a5dae0 100644
  --- a/builtin/mv.c
  +++ b/builtin/mv.c
  @@ -221,8 +221,11 @@ int cmd_mv(int argc, const char **argv, const char *prefix)
                                  }
                                  argc += last - first;
                          }
  -               } else if (!(ce = cache_file_exists(src, length, ignore_case))) {
  -                       bad = _("not under version control");
  +               } else if (!(ce = cache_file_exists(src, length, 0))) {
  +                       if (cache_file_exists(src, length, ignore_case))
  +                               bad = _("not under version control (upper/lower mixup)");
  +                       else
  +                               bad = _("not under version control");
                  } else if (ce_stage(ce)) {
                          bad = _("conflicted");
                  } else if (lstat(dst, &st) == 0 &&

===================
The exact message can be discussed.

@dscho @danmosemsft : Any thougths ?